app/main.py

```diff
@@ -1,25 +1,96 @@
+from abc import ABC, abstractmethod
 
 
 class IntegerRange:
-    pass
+    def __init__(self, min_amount: int, max_amount: int) -> None:
+        self.min_amount = min_amount
+        self.max_amount = max_amount
+
+    def __get__(self, instance: object, owner: object) -> int:
+        return instance.__dict__[self.name]
+
+    def __set__(self, instance: object, value: int) -> None:
+        if not self.min_amount <= value <= self.max_amount:
+            raise ValueError(f"{self.name} must "
+                             f"be between {self.min_amount} "
+                             f"and {self.max_amount}.")
+        instance.__dict__[self.name] = value
+
+    def __set_name__(self, owner: object, name: str) -> None:
+        self.name = name
 
 
 class Visitor:
-   pass
+    def __init__(
+            self,
+            name: str,
+            age: int,
+            weight: int,
+            height: int
+    ) -> None:
+        self.name = name
+        self.age = age
+        self.weight = weight
+        self.height = height
 
 
 class SlideLimitationValidator(ABC):
-    pass
+    def __init__(
+            self,
+            age_range: IntegerRange,
+            weight_range: IntegerRange,
+            height_range: IntegerRange
+    ) -> None:
+        self.age_range = age_range
+        self.weight_range = weight_range
+        self.height_range = height_range
+
+    @abstractmethod
+    def validate(self, visitor: Visitor) -> bool:
+        pass
 
 
 class ChildrenSlideLimitationValidator(SlideLimitationValidator):
-    pass
+    def __init__(self) -> None:
+        age_range = IntegerRange(4, 14)
+        height_range = IntegerRange(80, 120)
+        weight_range = IntegerRange(20, 50)
+        super().__init__(age_range, weight_range, height_range)
+
+    def validate(self, visitor: Visitor) -> bool:
+        return (
+            self.age_range.min_amount <= visitor.age
+            <= self.age_range.max_amount
+            and self.height_range.min_amount <= visitor.height
+            <= self.height_range.max_amount
+            and self.weight_range.min_amount <= visitor.weight
+            <= self.weight_range.max_amount
+        )
 
 
 class AdultSlideLimitationValidator(SlideLimitationValidator):
-    pass
+    def __init__(self) -> None:
+        age_range = IntegerRange(14, 60)
+        height_range = IntegerRange(120, 220)
+        weight_range = IntegerRange(50, 120)
+        super().__init__(age_range, weight_range, height_range)
+
+    def validate(self, visitor: Visitor) -> bool:
+        return (
+            self.age_range.min_amount <= visitor.age
+            <= self.age_range.max_amount
+            and self.height_range.min_amount <= visitor.height
+            <= self.height_range.max_amount
+            and self.weight_range.min_amount <= visitor.weight
+            <= self.weight_range.max_amount
+        )
 
 
 class Slide:
-    pass
+    def __init__(self, name: str, limitation_class: type) -> None:
```

Comment:
Specify 'limitation_class' as a list of 'SlideLimitationValidator' instances for clarity